diff --git a/node-postgres/session.cjs b/node-postgres/session.cjs
index 863d5762b3cd48b2045952c370d1217f84481fdb..dcbb5e40907c4676246af08cf912faf1244cb7b3 100644
--- a/node-postgres/session.cjs
+++ b/node-postgres/session.cjs
@@ -41,7 +41,8 @@ var import_session = require("../pg-core/session.cjs");
 var import_sql = require("../sql/sql.cjs");
 var import_tracing = require("../tracing.cjs");
 var import_utils = require("../utils.cjs");
-const { Pool, types } = import_pg.default;
+const { Pool, native, types } = import_pg.default;
+const NativePool = native?.Pool;
 class NodePgPreparedQuery extends import_session.PgPreparedQuery {
   constructor(client, queryString, params, logger, fields, name, _isResponseInArrayMode, customResultMapper) {
     super({ sql: queryString, params });
@@ -200,7 +201,9 @@ class NodePgSession extends import_session.PgSession {
     );
   }
   async transaction(transaction, config) {
-    const session = this.client instanceof Pool ? new NodePgSession(await this.client.connect(), this.dialect, this.schema, this.options) : this;
+    const session = (this.client instanceof Pool || (NativePool && this.client instanceof NativePool))
+      ? new NodePgSession(await this.client.connect(), this.dialect, this.schema, this.options)
+      : this;
     const tx = new NodePgTransaction(this.dialect, session, this.schema);
     await tx.execute(import_sql.sql`begin${config ? import_sql.sql` ${tx.getTransactionConfigSQL(config)}` : void 0}`);
     try {
@@ -211,7 +214,7 @@ class NodePgSession extends import_session.PgSession {
       await tx.execute(import_sql.sql`rollback`);
       throw error;
     } finally {
-      if (this.client instanceof Pool) {
+      if (this.client instanceof Pool || (NativePool && this.client instanceof NativePool)) {
         session.client.release();
       }
     }
